src/rescle.cc

```diff
@@ -1,890 +1,976 @@
-
-}  // namespace
-
-VersionInfo::VersionInfo(const HMODULE& hModule, const WORD& languageId)
-{
-  HRSRC hRsrc = FindResourceExW(hModule, RT_VERSION, MAKEINTRESOURCEW(1), languageId);
-
-  if (hRsrc == NULL) {
-    throw std::system_error(GetLastError(), std::system_category());
-  }
-
-  HGLOBAL hGlobal = LoadResource(hModule, hRsrc);
-  if (hGlobal == NULL) {
-    throw std::system_error(GetLastError(), std::system_category());
-  }
-
-  void* p = LockResource(hGlobal);
-  if (p == NULL) {
-    throw std::system_error(GetLastError(), std::system_category());
-  }
-
-  DWORD size = SizeofResource(hModule, hRsrc);
-  if (size == 0) {
-    throw std::system_error(GetLastError(), std::system_category());
-  }
-
-  DeserializeVersionInfo(static_cast<BYTE*>(p), size);
-}
-
-bool VersionInfo::HasFixedFileInfo() const
-{
-  return m_fixedFileInfo.dwSignature == 0xFEEF04BD;
-}
-
-VS_FIXEDFILEINFO& VersionInfo::GetFixedFileInfo()
-{
-  return m_fixedFileInfo;
-}
-
-void VersionInfo::SetFixedFileInfo(const VS_FIXEDFILEINFO& value)
-{
-  m_fixedFileInfo = value;
-}
-
-std::vector<BYTE> VersionInfo::Serialize()
-{
-  VersionStampValue versionInfo;
-  versionInfo.szKey = L"VS_VERSION_INFO";
-  versionInfo.wType = 0;
-
-  if (HasFixedFileInfo())
-  {
-    auto size = sizeof(VS_FIXEDFILEINFO);
-    versionInfo.wValueLength = size;
-    
-    auto& dst = versionInfo.Value;
-    dst.resize(size);
-
-    memcpy(&dst[0], &GetFixedFileInfo(), size);
-  }
-
-  {
-    VersionStampValue stringFileInfo;
-    stringFileInfo.szKey = L"StringFileInfo";
-    stringFileInfo.wType = 1;
-    stringFileInfo.wValueLength = 0;
-
-    auto& stringTables = StringTables;
-    for (auto iTable = stringTables.begin(); iTable != stringTables.end(); ++iTable)
-    {
-      VersionStampValue stringTableRaw;
-      stringTableRaw.wType = 1;
-      stringTableRaw.wValueLength = 0;
-
-      {
-        auto& translate = iTable->Encoding;
-        std::wstringstream ss;
-        ss << std::hex << std::setw(8) << std::setfill(L'0') << (translate.wLanguage << 16 | translate.wCodePage);
-        stringTableRaw.szKey = ss.str();
-      }
-
-      auto& strings = iTable->Strings;
-      for (auto iString = strings.begin(); iString != strings.end(); ++iString)
-      {
-        auto& stringValue = iString->second;
-        auto strLenNullTerminated = stringValue.length() + 1;
-
-        VersionStampValue stringRaw;
-        stringRaw.wType = 1;
-        stringRaw.szKey = iString->first;
-        stringRaw.wValueLength = strLenNullTerminated;
-
-        auto size = strLenNullTerminated * sizeof(WCHAR);
-        auto& dst = stringRaw.Value;
-        dst.resize(size);
-
-        auto src = stringValue.c_str();
-        
-        memcpy(&dst[0], src, size);
-
-        stringTableRaw.Children.push_back(std::move(stringRaw));
-      }
-
-      stringFileInfo.Children.push_back(std::move(stringTableRaw));
+﻿// Copyright (c) 2013 GitHub, Inc. All rights reserved.

```

Comment:
The changes in this file are really difficult to read. Did you change the line endings?